app/models/collection.rb

```diff
@@ -0,0 +1,8 @@
+# typed: strict
+class Collection < ApplicationRecord
+  has_many :items
+
+  # Define the attributes with their respective data types and constraints
+  validates :name, presence: true
+  validates :description, presence: true
+end
```

Comment:
The Collection model lacks an index on the 'name' attribute. Considering 'name' could be used for lookups, adding an index could improve performance. Suggest adding an index to the 'name' field in the corresponding migration file.